### PR TITLE
Add weekly hash check to NUTS file and XLSX to YAML utility function

### LINKED ISSUE #38: Updated source file by EUROSTAT
Looks like Eurostat silently updated the NUTS xlsx file, the hash is now different - and unit tests in other repositories are failing because of it...

https://github.com/IAMconsortium/legacy-definitions/actions/runs/11296344229/job/31421010160?pr=18

The feature with downloading the xlsx from Eurostat is nice, but risky if this happens from time to time without any announcement or quick solution for downstream users.

Maybe instead keeping a json or yaml of the relevant data in the repo is a more stable solution, like https://github.com/pycountry/pycountry/tree/main/src/pycountry/databases?

### PR DESCRIPTION
**Pull request by dc-almeida**
Closes #38
Adds a weekly GH action to compare the hash of a local NUTS file with the EUROSTAT website version
Adds a XLSX to YAML utility function for use when updating the local file to latest version

### CODE REVIEW

File: .github/workflows/check_hash.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Calculate Local File Hash
+        id: local-hash
+        run: echo "LOCAL_HASH=$(sha256sum pysquirrel/data/NUTS2021-NUTS2024.xlsx | awk '{print $1}')" >> $GITHUB_ENV
+
+      - name: Download File
+        run: curl -o most-recent-version.xlsx "https://ec.europa.eu/eurostat/documents/345175/629341/NUTS2021-NUTS2024.xlsx"
```
Comment (phackstock) [inline .github/workflows/check_hash.yml:19]: Does this actually always fetch the most recent version? Does the URL, either in the file name or the preceding address change? If so, this workflow would of course just always download the same file regardless of whether or not it's the latest NUTS file.

**Review by phackstock — COMMENTED** [review]
One general thought from my side. Now that you save a copy of the NUTS file as part of this repository, you don't really need the yaml files anymore. I'd suggest to do away with the yaml files and just read the Excel file directly. This way you also save the additional step of converting the Excel to yaml in case there is any update. And you exclude the error case where the Excel file and the yaml files are out of sync.
@danielhuppmann, what are your thoughts on the matter?
 In addition, one comment below in line.

**Comment by danielhuppmann** [comment]
First question: is the performance difference between an xlsx spreadsheet versus the yaml files noticeable?

Second question: if Eurostat again changes the file (hosted on the same url), we will only be notified that the hashes don't match, without any guidance on the actual change... Having the xlsx-to-yaml utility can be a useful way to find the change and/or check whether the relevant for us.

**Comment by phackstock** [comment]
> First question: is the performance difference between an xlsx spreadsheet versus the yaml files noticeable?

Technically, yaml should be faster than xlsx. However, in practice, reading data is orders of magnitude slower so the difference does not matter.

> Second question: if Eurostat again changes the file (hosted on the same url), we will only be notified that the hashes don't match, without any guidance on the actual change... Having the xlsx-to-yaml utility can be a useful way to find the change and/or check whether the relevant for us.

Ok, if Eurostat does not provide a changelog then I can see the value of comparing yaml files. We should still be careful though that the Excel file and the yaml files match. Otherwise we might get a wrong warning.

File: .github/workflows/check_hash.yml (hunk 1 above)
Comment (dc-almeida) [reply .github/workflows/check_hash.yml:19]: As far as I can tell, this is the URL that always points to the most recent version. This URL redirects to the full URL with the download (the full URL has a hash on the end that changes with each version, [that's why the previous pysquirrel version broke, as a new file was being downloaded and the pooch hash check failed](https://github.com/iiasa/pysquirrel/issues/38)).

Comment (phackstock) [reply .github/workflows/check_hash.yml:19]: Ah ok that's good then.

**Review by phackstock — APPROVED** [review]
Good to be merged from my side. 
The only thing that might be nice to add is a short markdown or rst file on how to update the NUTS regions in case the file changes again. It might seem straightforward now but in a year or two we are guaranteed to forget about it.

**Comment by dc-almeida** [comment]
> Ok, if Eurostat does not provide a changelog then I can see the value of comparing yaml files. We should still be careful though that the Excel file and the yaml files match. Otherwise we might get a wrong warning.

YAML files are also comparable in GitHub changelogs, whereas XLSX files are not, so it allows tracking changes to the regions.

**Review by danielhuppmann — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**